crates/unleash-edge-client-api/src/heartbeat.rs

```diff
@@ -65,14 +66,14 @@ async fn heartbeat(
     (
         StatusCode::ACCEPTED,
         Json(HeartbeatResponse {
-            edge_license_state: app_state.license_state,
+            edge_license_state: *app_state.license_state.read().await,
         }),
     )
 }
 
 #[derive(Clone)]
 pub struct HeartbeatState {
-    license_state: LicenseState,
+    license_state: Arc<RwLock<LicenseState>>,
```

Comment:
Another very annoying thing here - in theory we shouldn't need the Arc<RwLock wrapping at this point. We should be able to transform to a copy of LicenseState but between axum marking this as non async and tokio being a bit wild with locks I can't find a way to clone this here

...wouldn't be a problem if this was an AtomicU8 :laughing: